### DISCUSSION TITLE
ability_upgrades is null in match details

### DISCUSSION THREAD

**Discussion by maximen39** [SteamRE/SteamKit#953] [General]
Hello dear developers!
I'm working with practice match.
I met a problem, that ability_upgrades list field is null in player object in packet k_EMsgGCMatchDetailsResponse, BUT when i'm trying to request data from steam api https://api.steampowered.com/IDOTA2Match_570/GetMatchDetails/V001/?key=MY_KEY&match_id=MATCH_ID, from a public match ability_upgrades is present, but also this exact public match returns null ability_upgrades list field when i'm getting k_EMsgGCMatchDetailsResponse from SteamKit
![image](https://user-images.githubusercontent.com/21176163/102607289-b6709800-4130-11eb-9d01-7707386f2a76.png)
![image](https://user-images.githubusercontent.com/21176163/102607755-82e23d80-4131-11eb-95ce-957eeb34d742.png)

**Comment by voided** [comment]
Were you able to nethook any examples of the GC returning a valid value for `ability_upgrades`? It's certainly possible the GC and WebAPI could have different responses.

**Comment by maximen39** [comment]
> Were you able to nethook any examples of the GC returning a valid value for `ability_upgrades`? It's certainly possible the GC and WebAPI could have different responses.

I've never experienced getting valid value for ability_upgrades by nethooking the GC, but somehow in dota match statistics there are these ability upgrades in Dota2 Client.
![image](https://user-images.githubusercontent.com/21176163/102661029-b8644680-4184-11eb-8882-f81df70cbb94.png)
